zuul-core/src/main/java/com/netflix/zuul/netty/server/SocketAddressProperty.java

```diff
@@ -0,0 +1,173 @@
+ *     </ul>
+ *
+ *  <p>Note that the local IPv4 binds only work for {@code 127.0.0.1}, and not any other loopback addresses.  Currently,
+ *      all IP stack specific bind types only "prefer" a stack; it is up to the OS and the JVM to pick the the final
+ *      address.
+ */
+public final class SocketAddressProperty extends StringDerivedProperty<SocketAddress> {
+
+    public enum BindType {
+        /**
+         * Supports any IP stack, for a given port.  This is the default behaviour.  This also indicates that the
+         * caller doesn't prefer a given IP stack.
+         */
+        ANY,
+        /**
+         * Binds on IPv4 {@code 0.0.0.0} address.
+         */
+        IPV4_ANY(() -> InetAddress.getByAddress("0.0.0.0", new byte[] {0, 0, 0, 0})),
+        /**
+         * Binds on IPv6 {@code ::} address.
+         */
+        IPV6_ANY(() -> InetAddress.getByAddress("::", new byte[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0})),
+        /**
+         * Binds on any local address. This indicates that the caller doesn't prefer a given IP stack.
+         */
+        ANY_LOCAL(InetAddress::getLoopbackAddress),
+        /**
+         * Binds on the IPv4 {@code 127.0.0.1} localhost address.
+         */
+        IPV4_LOCAL(() -> InetAddress.getByAddress("localhost", new byte[] {127, 0, 0, 1})),
+        /**
+         * Binds on the IPv6 {@code ::1} localhost address.
+         */
+        IPV6_LOCAL(() ->
+                InetAddress.getByAddress("localhost", new byte[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1})),
+        /**
+         * Binds on the Unix Domain Socket path.
+         */
+        UDS,
+        ;
+
+        @Nullable
+        private final Supplier<? extends InetAddress> addressSupplier;
+
+        BindType() {
+            addressSupplier = null;
+        }
+
+        BindType(Callable<? extends InetAddress> addressFn) {
+            this.addressSupplier = () -> {
+                try {
+                    return addressFn.call();
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            };
+        }
+    }
+
+    @VisibleForTesting
+    static final class Decoder implements com.google.common.base.Function<String, SocketAddress> {
+
+        static final Decoder INSTANCE = new Decoder();
+
+        @Override
+        public SocketAddress apply(String input) {
+            if (input == null || input.isEmpty()) {
+                throw new IllegalArgumentException("Invalid address");
+            }
+
+            int equalsPosition = input.indexOf('=');
+            if (equalsPosition == -1) {
+                throw new IllegalArgumentException("Invalid address " + input);
+            }
+            String rawBindType = equalsPosition != 0 ? input.substring(0, equalsPosition) : BindType.ANY.name();
+            BindType bindType = BindType.valueOf(rawBindType.toUpperCase(Locale.ROOT));
```

Comment:
Why add Locale here?

Reply:
If not specific, the behavior changes based on the system you run on.  Picking ROOT forces the behavior to be consistent.  (for example, if for some reason in the future unicode was supported here).    Also, ErrorProne will nag if you don't.